src/core/coap/coap.hpp

```diff
@@ -396,6 +396,19 @@ class ResponsesQueue
         kMaxCachedResponses = OPENTHREAD_CONFIG_COAP_SERVER_MAX_CACHED_RESPONSES,
     };
 
+    /**
+     * Check if CoAP response exists in the cache that matches given Message ID and source endpoint.
+     *
+     * @param[in]  aRequest      The CoAP message containing Message ID.
+     * @param[in]  aMessageInfo  The message info containing source endpoint address and port.
+     * @param[out] aResponse     A pointer to the cached CoAP response matching given arguments, or NULL.
+     *
+     * @retval true   Matching response found.
+     * @retval false  Matching response not found.
+     *
+     */
+    bool FindMatchedResponse(const Message &aRequest, const Ip6::MessageInfo &aMessageInfo, Message **aResponse);
```

Comment:
-> `Message *FindMatchedResponse(const Message &aRequest, const Ip6::MessageInfo &aMessageInfo) const;`?

Reply:
Is this solved?

Reply:
yes

Reply:
I didn't see there's any change about the function prototype.

Reply:
Sorry, I thought you just meant to add const. 
Prototype modified now.